app/src/main/java/com/example/lexis/fragments/ArticleFragment.java

```diff
@@ -93,21 +87,24 @@ private SpannableStringBuilder styleTranslatedWords(Article article) {
         int curr = 0; // keep track of what index of the translated words we are on
         for (int i = 0; i < words.length; i++) {
             int start = spannableStringBuilder.length();
-            spannableStringBuilder.append(words[i]).append(" ");
             // TODO: some way to only do the translated words? other ones all have the same format
 
             // the index we're at represents a translated word
             if (curr < article.getTranslatedIndices().size() && article.getTranslatedIndices().get(curr) == i) {
-                final int translatedWordIndex = i;
                 final int originalWordIndex = curr;
 
+                // strip punctuation and store indices where letters in target word start/end
+                // so we don't highlight leading or trailing punctuation
+                String targetLanguage = words[i]; // already stripped of punctuation
+                String englishWithPunctuation = article.getOriginalWords().get(originalWordIndex);
+                int[] targetStartEnd = new int[2];
+                String english = Utils.stripPunctuation(englishWithPunctuation, targetStartEnd);
```

Comment:
Nice! So it was an issue with the punctuation then (the translation of corporate today)

Reply:
Yes! The weird characters like the "\&#39;" we saw in the translation of corporate were caused by escaped HTML characters, usually punctuation, in the response from the Google Translation API (for some reason, corporate is getting translated as "d'entreprise"). I approached the fix in two ways:

- Some words were having this issue because leading/trailing punctuation was getting included in the translation that shouldn't be (like the double quote getting included in the "Moi" in the screenshot above). I fixed this by stripping punctuation from the string before translating it.
- For some words like corporate, the punctuation was just getting included within the translation, so I used the Apache Commons StringEscapeUtils library to unescape the HTML so that these characters would show up correctly.